packages/apollo-federation/src/composition/__tests__/composeAndValidate.test.ts

```diff
@@ -551,3 +551,76 @@ describe('composition of value types', () => {
     });
   });
 });
+
+describe('composition of schemas with directives', () => {
+  /**
+   * To see which usage sites indicate whether a directive is "executable" or
+   * merely for use by the type-system ("type-system"), see the GraphQL spec:
+   * https://graphql.github.io/graphql-spec/June2018/#sec-Type-System.Directives
+   */
+  it('preserves executable and purges type-system directives', () => {
+    const serviceA = {
+      typeDefs: gql`
+        "directives at FIELDs are executable"
+        directive @audit(risk: Int!) on FIELD
+
+        "directives at FIELD_DEFINITIONs are for the type-system"
```

Comment:
It seems that `@deprecated` defies this semantic: https://www.apollographql.com/docs/graphql-tools/schema-directives/